cfme/tests/infrastructure/test_tenant_quota.py

```diff
@@ -316,3 +339,63 @@ def test_vm_migration_after_assigning_tenant_quota(appliance, small_vm, set_root
+
+
+# Args of parametrize is the list of navigation parameters to order catalog item.
+# first arg of parametrize is the list of fixtures or parameters,
+# second arg is a list of lists, with each one a test is to be generated
+# sequence is important here
+# indirect is the list where we define which fixtures are to be passed values indirectly.
+@pytest.mark.parametrize('context', [ViaSSUI, ViaUI])
+@pytest.mark.parametrize(
+    ['set_roottenant_quota', 'custom_prov_data', 'extra_msg', 'set_default'],
+    [
+        [('cpu', '2'), {'hardware': {'num_sockets': '8'}}, '', True],
+        [('storage', '0.01'), {}, '', True],
+        [('memory', '2'), {'hardware': {'memory': '4096'}}, '', True],
+        [('vm', '1'), {'catalog': {'num_vms': '4'}}, '###', True],
+    ],
+    indirect=['set_roottenant_quota', 'custom_prov_data', 'set_default'],
+    ids=['max_cpu', 'max_storage', 'max_memory', 'max_vms']
+)
+def test_service_infra_tenant_quota_with_default_entry_point(request, appliance, context,
+                                                             set_roottenant_quota, extra_msg,
+                                                             set_default, custom_prov_data,
+                                                             catalog_item):
+    """Test Tenant Quota in UI and SSUI by selecting field entry points.
+       Quota has to be checked if it is working with field entry points also.
+
+    Polarion:
+        assignee: ghubale
+        casecomponent: Infra
+        caseimportance: medium
+        initialEstimate: 1/12h
+        tags: quota
+        testSteps:
+            1. Add infrastructure provider
+            2. Set quota for root tenant - 'My Company'
+            3. Navigate to services > catalogs
+            4. Create catalog item with selecting following field entry points:
+                a.provisioning_entry_point = /ManageIQ (Locked)/Infrastructure/VM/Provisioning
+                /StateMachines/ProvisionRequestApproval/Default
+                b.retirement_entry_point = /Service/Retirement/StateMachines/ServiceRetirement
+                /Default
+            5. Add other information required in catalog for provisioning VM
+            6. Order the catalog item via UI and SSUI individually
+    """
+    with appliance.context.use(context):
+        service_catalogs = ServiceCatalogs(appliance, catalog_item.catalog, catalog_item.name)
+        if context is ViaSSUI:
+            service_catalogs.add_to_shopping_cart()
+        service_catalogs.order()
+    # nav to requests page to check quota validation
+    request_description = 'Provisioning Service [{name}] from [{name}]'.format(
+        name=catalog_item.name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui')
+
+    @request.addfinalizer
+    def delete():
+        provision_request.remove_request()
+        catalog_item.delete()
```

Comment:
Why not delete this inside the fixture that creates this catalog item?

Reply:
Good catch!! @valaparthvi